devel/gtk-osx-application/Portfile

```diff
@@ -61,116 +49,143 @@ dependencies are built with variants +quartz and try again.
   }
 }
 
-# py27-gtk-osx-application-gtk3 is not needed (only gir matters).
-# If a python gtk3 program requires gtk-osx-application, add
-#    depends_lib-append port:gtk-osx-application-gtk3 \
-#                       port:py27-gobject3
-
-subport py27-gtk-osx-application-gtk2 {
-    set gtk_version gtk2
-
-    depends_lib-append port:py27-pygtk \
-                       port:gtk-osx-application-gtk2
-    
-    set python_prefix           ${frameworks_dir}/Python.framework/Versions/2.7
+set python_version          3.7
+
+variant gtk2 requires python27 conflicts gtk3 description {Build with support for GTK+-2.0} {
+    global gtk_version
+    set gtk_version             gtk2
+    depends_lib-append          port:gtk2 \
+                                port:py27-pygtk \
+                                port:py27-gobject
+    require_active_variants     py27-gobject quartz
+    require_active_variants     py27-cairo quartz
+    require_active_variants     py27-pygtk quartz
+    require_active_variants     libglade2 quartz
+    configure.args-append       --with-gtk2 --without-gtk3 --enable-introspection=yes
+}
 
-    configure.python            ${python_prefix}/bin/python2.7
-    configure.env-append        PYGTK_CODEGEN=${python_prefix}/bin/pygtk-codegen-2.0
-    configure.env-append        PYGOBJECT_CODEGEN=${python_prefix}/bin/pygobject-codegen-2.0
-    configure.pkg_config_path   ${python_prefix}/lib/pkgconfig/
-    configure.pre_args          --prefix=${python_prefix}
+variant gtk3 conflicts gtk2 description {Build with support for GTK+-3.0} {
+    global gtk_version
+    set gtk_version             gtk3
+    depends_lib-append          port:gtk3
+    configure.args-append       --with-gtk3 --without-gtk2 --enable-introspection=yes
+}
 
+variant python27 conflicts python35 python36 python37 python38 description {Build bindings for Python 2.7} {
+    depends_lib-append          port:python27
+    require_active_variants     py27-cairo quartz
+}
 
-    configure.args-append --with-gtk2 --without-gtk3 --disable-introspection
+variant python35 requires gtk3 conflicts python27 python36 python37 python38 description {Build bindings for Python 3.5} {
+    depends_lib-append          port:python35
+}
```

Comment:
There is no need to add a variant for this old python version. Please remove it.

Reply:
I beg to differ here and in the following as well. While it's a fact they are old versions; they are still actively supported and there might be a valid reason for somebody to prefer using them (plus, it's not exactly a huge amount of work to provide users with the option; considering this port is targeted pretty much only at developers)

Reply:
That's fine as long as the maintainer agrees to it. If it's going to be nomaintainer, then I am against the added complexity that no one is willing to maintainer.